File: 3120. count-the-number-of-special-characters-i.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        lower = 0
        upper = 0

        for ch in word:
            if ch.islower():
                lower |= (1 << (ord(ch) - ord('a')))
            else:
                upper |= (1 << (ord(ch) - ord('A')))

        common = lower & upper

        # counting number of set bits
        return common.bit_count()

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    word1 = "aaAbcBC"
    output1 = sol.numberOfSpecialChars(word1)
    print(f"Example 1: word={word1}")
    print(f"Output: {output1}")
    print(f"Expected: 3")
    print()
    
    # Example 2
    word2 = "abc"
    output2 = sol.numberOfSpecialChars(word2)
    print(f"Example 2: word={word2}")
    print(f"Output: {output2}")
    print(f"Expected: 0")
    print()
    
    # Example 3
    word3 = "abBCab"
    output3 = sol.numberOfSpecialChars(word3)
    print(f"Example 3: word={word3}")
    print(f"Output: {output3}")
    print(f"Expected: 1")

File: 3120. count-the-number-of-special-characters-i.java
class Solution {
    public int numberOfSpecialChars(String word) {
        int lower = 0;
        int upper = 0;

        for(char ch : word.toCharArray()) {
            if(Character.isLowerCase(ch)) {
                lower |= (1 << (ch - 'a'));
            }
            else {
                upper |= (1 << (ch - 'A'));
            }
        }

        int common = lower & upper;

        // counting number of set bits
        return Integer.bitCount(common);
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        String word1 = "aaAbcBC";
        int output1 = sol.numberOfSpecialChars(word1);
        System.out.println("Example 1: word=\"aaAbcBC\"");
        System.out.println("Output: " + output1);
        System.out.println("Expected: 3");
        System.out.println();
        
        // Example 2
        String word2 = "abc";
        int output2 = sol.numberOfSpecialChars(word2);
        System.out.println("Example 2: word=\"abc\"");
        System.out.println("Output: " + output2);
        System.out.println("Expected: 0");
        System.out.println();
        
        // Example 3
        String word3 = "abBCab";
        int output3 = sol.numberOfSpecialChars(word3);
        System.out.println("Example 3: word=\"abBCab\"");
        System.out.println("Output: " + output3);
        System.out.println("Expected: 1");
    }
}